src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_Diagnostics.cs

```diff
@@ -31,25 +31,25 @@ public Task CalculateDiagnosticsAsync(DiagnosticArguments arguments, string stre
                 using (RoslynLogger.LogBlock(FunctionId.CodeAnalysisService_CalculateDiagnosticsAsync, arguments.ProjectId.DebugName, cancellationToken))
                 using (arguments.ForcedAnalysis ? UserOperationBooster.Boost() : default)
                 {
-                    try
-                    {
-                        // entry point for diagnostic service
-                        var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
+                    // entry point for diagnostic service
+                    var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
 
-                        var optionSet = await RoslynServices.AssetService.GetAssetAsync<OptionSet>(arguments.OptionSetChecksum, cancellationToken).ConfigureAwait(false);
-                        var projectId = arguments.ProjectId;
-                        var analyzers = RoslynServices.AssetService.GetGlobalAssetsOfType<AnalyzerReference>(cancellationToken);
+                    var optionSet = await RoslynServices.AssetService.GetAssetAsync<OptionSet>(arguments.OptionSetChecksum, cancellationToken).ConfigureAwait(false);
+                    var projectId = arguments.ProjectId;
+                    var analyzers = RoslynServices.AssetService.GetGlobalAssetsOfType<AnalyzerReference>(cancellationToken);
 
-                        var result = await (new DiagnosticComputer(solution.GetProject(projectId))).GetDiagnosticsAsync(
-                            analyzers, optionSet, arguments.AnalyzerIds, arguments.ReportSuppressedDiagnostics, arguments.LogAnalyzerExecutionTime, cancellationToken).ConfigureAwait(false);
+                    var result = await new DiagnosticComputer(solution.GetProject(projectId)).GetDiagnosticsAsync(
+                        analyzers, optionSet, arguments.AnalyzerIds, arguments.ReportSuppressedDiagnostics, arguments.LogAnalyzerExecutionTime, cancellationToken).ConfigureAwait(false);
 
-                        await SerializeDiagnosticResultAsync(streamName, result, cancellationToken).ConfigureAwait(false);
-                    }
-                    catch (IOException)
+                    await ClientDirectStream.WriteDataAsync(streamName, result, (writer, data, cancellationToken) =>
```

Comment:
so here we write synchronously to the stream.  Are there cases where we actually need to be able to write asynchronously?  if not, consider making it so that hte delegate it take in returns void and not a task.

Reply:
It's removed in subsequent commit.